tools/distpartitioning/dataset_utils.py

```diff
@@ -337,11 +337,18 @@ def get_dataset(input_dir, graph_name, rank, world_size, schema_map):
             if not os.path.isabs(edge_file):
                 edge_file = os.path.join(input_dir, edge_file)
             logging.info(f'Loading edges of etype[{etype_name}] from {edge_file}')
-            data_df = csv.read_csv(edge_file,
-                        read_options=pyarrow.csv.ReadOptions(autogenerate_column_names=True), 
-                        parse_options=pyarrow.csv.ParseOptions(delimiter=' '))
-            src_ids.append(data_df['f0'].to_numpy())
-            dst_ids.append(data_df['f1'].to_numpy())
+
+            read_options=pyarrow.csv.ReadOptions(use_threads=True, block_size=4096, autogenerate_column_names=True)
```

Comment:
how to define best value for `block_size`? have you profiled perf with different values? `4096` is the default block size in modern computer?

Reply:
`4096` is a bit small, I think.

Reply:
Since we are using multiple threads, I guess internally in pyarrow each thread is allowed to a chunk of the size 4096. If we do not perform chunked reading of the file.. it appears that the pyarrow is taking massive amounts of RAM to complete the reading csv files. But at the same time it is very fast, that fastest I have run so far. 

Even after reading is complete and pyarrow variables becomes out of scope, the memory consumed for reading files was NOT getting released back to the system. 

To decrease the memory foot print of pyarrow library module when reading csv files, we now use chunked reading of the files. This is reduced the total amount of memory used by pyarrow very significantly, to the order of 50GB+ for fe_large graph. 

We can use various chunk sizes, 4096, 4096*2, 4096*4 etc... Of all these block sizes I did not notice any significant change in the time to read the csv files. There is a very slight change the resident memory (RAM) used in all these instances.